libshvchainpack/src/chainpack/rpcvalue.cpp

```diff
@@ -165,13 +165,25 @@ class ValueData : public RpcValue::AbstractValueData
 	RpcValue::MetaData *m_metaData = nullptr;
 };
 
+namespace {
+bool is_representable_as_int(const double d)
+{
+	return std::numeric_limits<RpcValue::Int>::max() >= d && std::numeric_limits<RpcValue::Int>::min() <= d;
+}
+}
+
 class ChainPackDouble final : public ValueData<RpcValue::Type::Double, double>
 {
 	ChainPackDouble* create() override { return new ChainPackDouble(0); }
 	std::string toStdString() const override { return std::to_string(m_value); }
 	double toDouble() const override { return m_value; }
 	bool toBool() const override { return !(m_value == 0.); }
-	RpcValue::Int toInt() const override { return static_cast<RpcValue::Int>(m_value); }
+	RpcValue::Int toInt() const override {
+		if (!is_representable_as_int(m_value)) {
+			throw std::runtime_error("This double is not representable as int");
```

Comment:
I'm not sure, if I adding `throw` to `to*()` is good idea, what about return `nan`